### PR TITLE
Prize limit for Arcades

### PR DESCRIPTION
**Pull request by Jackrost**
This PR limit prizes for Space Villain arcade. All arcades on maps (static and spawner) were changed. No more arcade abuse

**Warning for mappers**
If you want to add Space Villain arcade on map - use SpaceVillainArcadeFilled

:cl:
- tweak: Space Villain arcade prizes now limited. Number of prizes are randomized

### CODE REVIEW

File: Content.Server/Arcade/Components/SpaceVillainArcadeComponent.cs [hunk 1]
```diff
@@ -115,8 +119,12 @@ private void UserInterfaceOnOnReceiveMessage(ServerBoundUserInterfaceMessage ser
         /// </summary>
         public void ProcessWin()
         {
-            var entityManager = IoCManager.Resolve<IEntityManager>();
-            entityManager.SpawnEntity(_random.Pick(_possibleRewards), entityManager.GetComponent<TransformComponent>(Owner).MapPosition);
+            if (_rewardAmount > 0)
+            {
+                var entityManager = IoCManager.Resolve<IEntityManager>();
+                entityManager.SpawnEntity(_random.Pick(_possibleRewards), entityManager.GetComponent<TransformComponent>(Owner).MapPosition);
```
Comment (EmoGarbage404) [inline Content.Server/Arcade/Components/SpaceVillainArcadeComponent.cs:125]:
```suggestion
                _entityManager.SpawnEntity(_random.Pick(_possibleRewards), entityManager.GetComponent<TransformComponent>(Owner).MapPosition);
```

File: Content.Server/Arcade/Components/SpaceVillainArcadeComponent.cs [hunk 2]
```diff
@@ -115,8 +119,12 @@ private void UserInterfaceOnOnReceiveMessage(ServerBoundUserInterfaceMessage ser
         /// </summary>
         public void ProcessWin()
         {
-            var entityManager = IoCManager.Resolve<IEntityManager>();
-            entityManager.SpawnEntity(_random.Pick(_possibleRewards), entityManager.GetComponent<TransformComponent>(Owner).MapPosition);
+            if (_rewardAmount > 0)
+            {
+                var entityManager = IoCManager.Resolve<IEntityManager>();
```
Comment (EmoGarbage404) [inline Content.Server/Arcade/Components/SpaceVillainArcadeComponent.cs:124]:
```suggestion
```

File: Content.Server/Arcade/Components/SpaceVillainArcadeComponent.cs [hunk 3]
```diff
@@ -63,6 +63,10 @@ public sealed class SpaceVillainArcadeComponent : SharedSpaceVillainArcadeCompon
             "HarmonicaInstrument", "OcarinaInstrument", "RecorderInstrument", "GunpetInstrument", "BirdToyInstrument"
         };
 
+        [ViewVariables(VVAccess.ReadWrite)]
+        [DataField("rewardAmount")]
+        private int _rewardAmount;
```
Comment (EmoGarbage404) [inline Content.Server/Arcade/Components/SpaceVillainArcadeComponent.cs:68]: just make this public

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]
you probably don't need to have filled and non-filled variants.

**Comment by Jackrost** [comment]
> you probably don't need to have filled and non-filled variants.
But who fill them? Mapers? Big amount arcades created with spawners also.

File: Content.Server/Arcade/Components/SpaceVillainArcadeComponent.cs (hunk 1 above)
Comment (Jackrost) [reply Content.Server/Arcade/Components/SpaceVillainArcadeComponent.cs]: done

File: Content.Server/Arcade/Components/SpaceVillainArcadeComponent.cs (hunk 2 above)
Comment (Jackrost) [reply Content.Server/Arcade/Components/SpaceVillainArcadeComponent.cs]: done

File: Content.Server/Arcade/Components/SpaceVillainArcadeComponent.cs (hunk 3 above)
Comment (Jackrost) [reply Content.Server/Arcade/Components/SpaceVillainArcadeComponent.cs]: done

**Comment by Jackrost** [comment]
The main problem that construction doesn't allow me to set prize amount after completion. It's some sort of workaround

**Comment by Jackrost** [comment]
Fix it. Now number of prizes are randomised

**Comment by Jackrost** [comment]
It's don't conflict now, but you need to manually check and swap arcades to new ones if they exist

File: Content.Server/Arcade/Components/SpaceVillainArcadeComponent.cs [hunk 4]
```diff
@@ -115,8 +128,11 @@ private void UserInterfaceOnOnReceiveMessage(ServerBoundUserInterfaceMessage ser
         /// </summary>
         public void ProcessWin()
         {
-            var entityManager = IoCManager.Resolve<IEntityManager>();
-            entityManager.SpawnEntity(_random.Pick(_possibleRewards), entityManager.GetComponent<TransformComponent>(Owner).MapPosition);
+            if (_rewardAmount > 0)
+            {
+                _entityManager.SpawnEntity(_random.Pick(_possibleRewards), _entityManager.GetComponent<TransformComponent>(Owner).MapPosition);
```
Comment (AJCM-git) [inline Content.Server/Arcade/Components/SpaceVillainArcadeComponent.cs:133]: use entity coordinates (`<TransformComponent>(Owner).Coordinates` )

**Review by AJCM-git — CHANGES_REQUESTED** [review]
You can fix origin with git CLI with these commands
`git checkout arcade-limitation`
`git checkout master Resources/Maps/origin.yml`
Do a commit, then merge latest master to this branch and finally make sure that there arent any new `SpaceVillainArcade` in any of the maps

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**